homeassistant/components/doorbird/__init__.py

```diff
@@ -187,126 +156,67 @@ def custom_url(self):
         """Get custom url for device."""
         return self._custom_url
 
-    def update_schedule(self, hass):
-        """Register monitored sensors and deregister others."""
-        from doorbirdpy import DoorBirdScheduleEntrySchedule
-
-        # Create a new schedule (24/7)
-        schedule = DoorBirdScheduleEntrySchedule()
-        schedule.add_weekday(0, 604800)  # seconds in a week
+    @property
+    def token(self):
+        return self._token
 
+    def register_events(self, hass):
         # Get the URL of this server
         hass_url = hass.config.api.base_url
 
         # Override url if another is specified in the configuration
         if self.custom_url is not None:
             hass_url = self.custom_url
 
-        # For all sensor types (enabled + disabled)
-        for sensor_type in SENSOR_TYPES:
-            name = '{} {}'.format(self.name, SENSOR_TYPES[sensor_type]['name'])
-            slug = slugify(name)
-
-            url = '{}{}/{}?token={}'.format(hass_url, API_URL, slug,
-                                            self._token)
-            if sensor_type in self._monitored_events:
-                # Enabled -> register
-                self._register_event(url, sensor_type, schedule)
-                _LOGGER.info('Registered for %s pushes from DoorBird "%s". '
-                             'Use the "%s_%s" event for automations.',
-                             sensor_type, self.name, DOMAIN, slug)
-
-                # Register a dummy listener so event is listed in GUI
-                hass.bus.listen('{}_{}'.format(DOMAIN, slug), handle_event)
-            else:
-                # Disabled -> deregister
-                self._deregister_event(url, sensor_type)
-                _LOGGER.info('Deregistered %s pushes from DoorBird "%s". '
-                             'If any old favorites or schedules remain, '
-                             'follow the instructions in the component '
-                             'documentation to clear device registrations.',
-                             sensor_type, self.name)
-
-    def _register_event(self, hass_url, event, schedule):
-        """Add a schedule entry in the device for a sensor."""
-        from doorbirdpy import DoorBirdScheduleEntryOutput
+        for event in self._events:
+            event = self._get_event_name(event)
 
-        # Register HA URL as webhook if not already, then get the ID
-        if not self.webhook_is_registered(hass_url):
-            self.device.change_favorite('http', 'Home Assistant ({} events)'
-                                        .format(event), hass_url)
+            self._register_event(hass_url, event)
 
-        fav_id = self.get_webhook_id(hass_url)
+            # Register a dummy listener so event is listed in GUI
+            hass.bus.listen('{}_{}'.format(DOMAIN, event), handle_event)
```

Comment:
If we're not acting on the event we shouldn't listen for the event on the event bus.

Please explain the idea behind this, so we can think of the correct approach.

Reply:
The idea behind registering a dummy listener is for user convenience of identifying the events that were registered (via dev-event panel) and the correct names to be used in their automations.  The component doesn't need to take action on an event, but the events will need to be used in the automations.  This seemed to be the best approach for exposing this information to the user.

Reply:
Ok, thanks! We shouldn't register a dummy listener. I agree that it would be good for the user to be able to see events that can be fired. But that needs to be solved separately.